features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/AlarmProviderImpl.java

```diff
@@ -40,12 +47,17 @@
     private AlarmDao alarmDao;
 
     @Override
-    public AlarmWrapper lookup(String reductionKey) {
-        final OnmsAlarm alarm = alarmDao.findByReductionKey(reductionKey);
-        if (alarm == null) {
-            return null;
+    public Map<String, AlarmWrapper> lookup(Set<String> reductionKeys) {
+        if (reductionKeys == null || reductionKeys.isEmpty()) {
+            return new HashMap<>();
+        }
+        if (reductionKeys.size() <= 1000) {
```

Comment:
Can we move the `1000` to a constant, and maybe make it configurable via a system property? Similar to what we did with the `pollInterval`: https://github.com/OpenNMS/opennms/blob/develop/features/bsm/daemon/src/main/java/org/opennms/netmgt/bsm/daemon/Bsmd.java#L153